### DISCUSSION TITLE
How to reliably execute a script if a monitor is added or removed?

### DISCUSSION THREAD

**Discussion by coderPaddyS** [hyprwm/Hyprland#5644] [Q&A]
Hey all,
i've created a custom bar using eww and adapted it already such that it will displayed on each monitor if a script is run.
Now I want to run this sript every time a monitor is connected or disconnected, ideally also after suspension triggered by the lid switch.

My initial try was listening to the `.socket.2` event in a script started on login and after the lid switch is off:
```nu
socat -u $"UNIX-CONNECT:/tmp/hypr/($env.HYPRLAND_INSTANCE_SIGNATURE)/.socket2.sock" - | each {|line|
    if ($line | is-empty) {
        null
    } else {
        let event = $line | parse "{name}>>{target}" | first
        if ($event.name == monitorremoved or $event.name == monitoradded) {
            exec $"($env.HOME)/.config/eww/launch_bar"
        }
    }
}
```
However the socket-connection is terminated immediately after the workspaces reload and resort them.
Kanshi would allow me to run the script if my profile changes, but does not allow me directly to observe if a monitor is disconnected. This solution would also require to create a profile for any possible combination of monitors I would encounter and exec the script in each profile.
Does anybody has an idea?

**Comment by ordy** [comment]
udev rule? 

Can try to grab drm events on your card and either run a script directly or use a single-run systemd service to start your script. Then you could check for the monitor status in /sys/class/drm if you need to perform actions based on specific monitors.

Run `udevadm monitor -s drm -p` to see if you get udev events, there should be one for lid close events too.

**Comment by dr0bz** [comment]
Hi,
here is an example: https://gist.github.com/dr0bz/4f8bf8b8829506cf26d4e2acc770e5af
I check here if a monitor configuration has changed and adjust the waybar config, to show the bar on one or another output.

Regards
